src/docs/howto-listing/bookings-with-buffer/index.md

```diff
@@ -0,0 +1,411 @@
+- the function passed to _findBookingUnitBoundaries_ as _nextBoundaryFn_
+  by default is _findNextBoundary_, which is what we need to modify
+  first. The _findNextBoundary_ function increments the current boundary
+  by a predefined value.
+
+```js
+export const findNextBoundary = (timeZone, currentMomentOrDate) =>
+  moment(currentMomentOrDate)
+    .clone()
+    .tz(timeZone)
+    .add(1, 'hour') // The default handling uses hours
+    .startOf('hour') // By default, the time slot is rounded to the start of the hour
+    .toDate();
+```
+
+### Add a custom rounding function for moment.js
+
+FTW-hourly uses the [moment-timezone](https://momentjs.com/timezone/)
+library to modify times and dates and convert them between the listing's
+time zone and the user's time zone.
+
+By default, the _findNextBoundary_ function uses
+_moment.startOf('hour')_ to round the booking slots to the top of each
+hour. However, since we are now dealing with minutes, we need to create
+a custom rounding function to replace the _startOf('hour')_ function
+call. When we add it to _moment.js_ using the prototype exposed through
+_moment.fn_, we can chain it in the same place as the default
+_startOf('hour')_ function.
+
+This rounding function rounds to sharp hours when the buffer minutes
+value is a factor of an hour, e.g. 15, 20 or 30 minutes.
+
+```js
+/**
+ * Rounding function for moment.js. Rounds the Moment provided by the context
+ * to the start of the specified time value in the specified units.
+ * @param {*} value the rounding value
+ * @param {*} unit time units to specify the value
+ * @returns Moment rounded to the start of the specified time value
+ */
+moment.fn.startOfDuration = function(value, unit) {
+  const ms = moment.duration(value, unit)._milliseconds;
+  return moment(Math.floor(this / ms) * ms);
+};
+```
+
+You will then need to use the new function to replace the built-in
+_startOf()_ function.
+
+We also need to calculate the increment of time to add to each time
+boundary, i.e. how long are the stretches of time delineated by the
+boundaries. To do that, we need an _isStart_ attribute, i.e. whether
+we're dealing with start times or end times.
+
+We also need isFirst, i.e. whether the boundary in question is the very
+first one in the list. Since we're rounding to the buffer time (here: 15
+minutes), we'll need to manually set the first time slot to correspond
+to the start of the available time slot.
+
+```diff
+- export const findNextBoundary = (timeZone, currentMomentOrDate) =>
+-   moment(currentMomentOrDate)
++ export const findNextBoundary = (
++   timeZone,
++   currentMomentOrDate,
++   isFirst,
++   isStart
++ ) => {
++   const increment = !isStart ? hourMinutes // For end time slots, add a full hour
++     : isFirst ? 0             // For the first start slot, use the actual start time
++     : bufferMinutes;          // For other start slots, use the buffer time
++
++   return moment(currentMomentOrDate)
```

Comment:
```suggestion
+   const increment = !isStart 
+     ? hourMinutes // For end time slots, add a full hour
+     : isFirst 
+     ? 0             // For the first start slot, use the actual start time
+     : bufferMinutes;          // For other start slots, use the buffer time
+
+   return moment(currentMomentOrDate)
```

Conditional/ternary chain: I think it's better to teach the format that Prettier and our code uses:

```js
const value = hasX
  ? x
  : hasY
  ? y
  : hasZ
  ? z
  : fallback;